src/core/io/writer/h5md/h5md_core.hpp

```diff
@@ -39,180 +38,187 @@
+   * @return TRUE if H5MD structure is present, FALSE else.
+   */
+  bool check_for_H5MD_structure(std::string const &filename);
+  /**
+   * @brief Method that performs all the low-level stuff for writing the
+   * particle
+   * positions to the dataset.
+   */
+  template <typename T>
+  void WriteDataset(T &data, const std::string &path, int *change_extent,
+                    hsize_t *offset, hsize_t *count);
+
+  /**
+   * @brief Method that extends datasets by the given extent.
+   */
+  void ExtendDataset(std::string path, int *change_extent);
+
+  /**
+  * @brief Method that returns chunk dimensions.
+  */
+  std::vector<hsize_t> create_chunk_dims(hsize_t dim, hsize_t size,
+                                         hsize_t chunk_size);
+
+  /*
+   * @brief Method to fill the arrays that are used by WriteDataset particle by
+   * particle.
+   */
+  void fill_arrays_for_h5md_write_with_particle_property(
+      int particle_index, int_array_3d &id, int_array_3d &typ,
+      double_array_3d &mass, double_array_3d &pos, int_array_3d &image,
+      double_array_3d &vel, double_array_3d &f, double_array_3d &charge,
+      Particle *current_particle, int write_dat, int_array_3d &bond);
+  /*
+   * @brief Method to write the simulation script to the dataset.
+   */
+  void WriteScript(std::string const &filename);
+
+  /**
+   * @brief Creates a new H5MD file.
+   * @param filename The filename
+   */
+  void create_new_file(const std::string &filename);
+
+  /**
+   * @brief Loads an existing H5MD file.
+   * @param filename The filename
+   */
+  void load_file(const std::string &filename);
+
+  /**
+   * @brief Initializes the necessary data to create the datsets and attributes.
+   */
+  void init_filestructure();
+
+  /**
+   * @brief Creates the necessary HDF5 datasets.
+   * @param only_load Set this to true if you want to append to an existing
+   * file.
+   */
+  void create_datasets(bool only_load);
+
+  /**
+   * @brief Links the time and step datasets to of all properties to the time
+   * and step dataset of the id property. All properties are written at the same
+   * time.
+   */
+  void create_links_for_time_and_step_datasets();
+
+  /**
+   * Member variables.
+   */
+  int m_max_n_part = 0;
+  std::string m_filename;
+  std::string m_scriptname;
+  int m_what;
+  bool m_write_ordered;
+  std::string m_backup_filename;
+  boost::filesystem::path m_absolute_script_path = "NULL";
+  h5xx::file m_h5md_file;
+
+  struct DatasetDescriptor {
+    std::string path;
+    hsize_t dim;
+    h5xx::datatype type;
+  };
+  std::vector<std::string> group_names;
+  std::vector<DatasetDescriptor> dataset_descriptors;
+  std::unordered_map<std::string, h5xx::dataset> datasets;
 };
 
-
 struct incompatible_h5mdfile : public std::exception {
-    const char* what () const throw ()
-    {
-        return "The given hdf5 file does not have a valid h5md structure!";
-    }
+  const char *what() const throw() {
```

Comment:
Isn't "throw()" already telling the compiler that this function does not throw anything, therefore "noexcept" is not needed here?